Vacms 16208 police testing

eselkin opened the pull request:
## Summary

- Added unit testing to verify conversion of CSV data to JSON per expected format 
- Sitewide Facilities

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16208

## Testing done

- Adds testing for expected format of CSV (requested from Axon)
- Adds unit test with missing files (missing contacts data)
- Adds unit test with present data and expects conversion 
- Adds mock data in CSV format expected from Axon.

## Screenshots

No UI changes

## What areas of the site does it impact?

Solely unit testing is added.

## Acceptance criteria

unit tests pass

### Quality Assurance & Testing

- [x] I added unit tests for conversion of expected Axon CSV data.
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated (not necessary)
- [x] Screenshot of the developed feature is added (no screenshot)
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user (no changes for any path)

## Requested Feedback

Check that tests pass.

eselkin commented on src/site/stages/build/drupal/tests/police/download.unit.spec.js:
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @department-of-veterans-affairs/axe-check-required */
```
Unit test is not a e2e test

eselkin commented on src/site/stages/build/drupal/tests/police/download.unit.spec.js:
```diff
@@ -0,0 +1,75 @@
+  it('should have an error if the police-contact.csv file is missing', async () => {
+    const client = getCurlClient({
+      'drupal-user': 'eli.selkin@va.gov',
+      'drupal-password': 'drupal8',
```
Used in the test, because testing does not have access to build credentials.
Does not actually log in or use credential for test, just needs them to be present.

jilladams commented:
@eselkin this PR auto-tagged Public Websites front end, but not Facilities front-end. Could you add a rule to content-build `codeowners` file that makes Facilities the point of contact for the Police path? That can go into this PR, or a separate one, whatever is easier.

eselkin commented:
@jilladams It doesn't want to seem to catch that directory. Already we should be requested because we changed `src/site/stages/build/drupal` since the `vaPoliceData` directory is under there and we are not requested. I added it to be specific but GitHub doesn't request the review.

randimays commented on src/site/stages/build/drupal/static-data-files/fetchApi.js:
```diff
@@ -58,7 +58,6 @@ function getCurlClient(buildOptions, _clientOptionsArg = { verbose: true }) {
           'certs/VA-Internal-S2-RCA2.pem',
         ]);
       }
-
```
Nitpick **Consider**: extra lines before `return`s, loops and between variable definitions and procedural code tend to increase readability. Could we leave in the spacing? Definitely not blocking, just something to consider.

eselkin replied:
Yeah, I think that was the linter removing the space, let me see if I can add it back.

eselkin replied:
Nope, can't add the spaces, it errors the CI

maxx1128 replied:
I agree with Randi on this. I've tried to do the same thing and add extra lines for readability, but the linter changes it. This may be a separate ticket for adjusting the linter.
